server/sonar-server/src/main/java/org/sonar/server/rule/ws/SearchAction.java

```diff
@@ -555,4 +569,14 @@ public RuleKey apply(@Nonnull Rule input) {
       return input.key();
     }
   }
+
+  private enum TypeToString implements Function<RuleType, String> {
+    INSTANCE;
+
+    @Override
+    public String apply(@Nonnull RuleType input) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The "input" parameter nullability is different in the superclass method, and that should not be changed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)